src/main/java/com/novi/backend_spring_boot_tech_it_easy_controller/services/TelevisionService.java

```diff
@@ -0,0 +1,143 @@
+        return convertToDto(television);  // Return het bijgewerkte DTO
+    }
+
+    private RemoteControllerDto convertToRemoteControllerDto(RemoteController remoteController) {
+        RemoteControllerDto dto = new RemoteControllerDto();
+        dto.setId(remoteController.getId());
+        dto.setName(remoteController.getName());
+        dto.setBrand(remoteController.getBrand());
+        dto.setPrice(remoteController.getPrice());
+        return dto;
+    }
+
+    public TelevisionDto createTelevision(TelevisionInputDto inputDto) {
+        Television television = new Television();
+        television.setName(inputDto.getName());
+        television.setBrand(inputDto.getBrand());
+        television.setPrice(inputDto.getPrice());
+        television.setAvailableSize(inputDto.getAvailableSize());
+        television.setRefreshRate(inputDto.getRefreshRate());
+        television.setScreenType(inputDto.getScreenType());
+        television.setScreenQuality(inputDto.getScreenQuality());
+        television.setSmartTv(inputDto.getSmartTv());
+        television.setWifi(inputDto.getWifi());
+        television.setVoiceControl(inputDto.getVoiceControl());
+        television.setHdr(inputDto.getHdr());
+        television.setBluetooth(inputDto.getBluetooth());
+        television.setAmbiLight(inputDto.getAmbiLight());
+        television.setOriginalStock(inputDto.getOriginalStock());
+        television.setSold(inputDto.getSold());
+
+        television = televisionRepository.save(television);
+        return convertToDto(television);
+    }
+
+    private TelevisionDto convertToDto(Television television) {
+        TelevisionDto dto = new TelevisionDto();
+        dto.setId(television.getId());
+        dto.setName(television.getName());
+        dto.setBrand(television.getBrand());
+        dto.setPrice(television.getPrice());
+        dto.setAvailableSize(television.getAvailableSize());
+        dto.setRefreshRate(television.getRefreshRate());
+        dto.setScreenType(television.getScreenType());
+        dto.setScreenQuality(television.getScreenQuality());
+        dto.setSmartTv(television.getSmartTv());
+        dto.setWifi(television.getWifi());
+        dto.setVoiceControl(television.getVoiceControl());
+        dto.setHdr(television.getHdr());
+        dto.setBluetooth(television.getBluetooth());
+        dto.setAmbiLight(television.getAmbiLight());
+        dto.setOriginalStock(television.getOriginalStock());
+        dto.setSold(television.getSold());
+
+        // Voeg de RemoteController toe aan het Television DTO als deze bestaat
+        if (television.getRemoteController() != null) {
+            RemoteControllerDto remoteControllerDto = convertToRemoteControllerDto(television.getRemoteController());
+            dto.setRemoteController(remoteControllerDto);
+        }
+
+        return dto;
+    }
+
+    public void deleteTelevision(Long id) {
+        boolean exists = televisionRepository.existsById(id);
```

Comment:
Je kan hier ook de findbyid gebruiken en dan gelijk een throw else er op gooien

Reply:
ExistsById is hier heel goed gevonden en precies toepasselijk voor deze situatie